### PR TITLE
fix CI for PRs from forks

### PR DESCRIPTION
**Pull request by joemfb**
By only updating cachix if credentials are available.

### CODE REVIEW

**Comment by joemfb** [comment]
Also disables cross-compilation for PRs from forks, as that's untenable without build caching.

File: sh/ci-builds [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -ex
+
+make
+
+#  only cross-compile if cachix is available
```
Comment (benjamin-tlon) [inline sh/ci-builds]: Why? This is a public cache, so we can still pull things from the cache without this variables. We just need them in order to write to the cache.

File: sh/cachix [hunk 2]
```diff
@@ -6,7 +6,7 @@ fail () {
 }
 
 if [ -z "$CACHIX_SIGNING_KEY" ]
-then fail "The CACHIX_AUTH_TOKEN environment variable needs to be set."
```
Comment (benjamin-tlon) [inline sh/cachix]: I believe the `cachix authtoken` and `cachix use` commands are still necessary?

Comment (joemfb) [reply sh/cachix]: They're still in this file. I've removed the `authtoken` command from `.travis.yml`, but kept `use`.

File: sh/ci-builds (hunk 1 above)
Comment (joemfb) [reply sh/ci-builds]: My bad -- I removed the cachix commands from `.travis.yml` because they're also in `sh/cachix`, but that broke public cache use.

**Comment by joemfb** [comment]
Thanks for the feedback, this was way too heavy-handed. This PR is now just disabling push-to-cachix for PRs from forks and fixing a printf. PTAL.

**Comment by joemfb** [comment]
@ixv bump for review

**Pull request merged**